test/exdatatest.c

```diff
@@ -12,35 +12,42 @@
 #include <stdlib.h>
 #include <openssl/crypto.h>
 
+#include "test_main.h"
+#include "testutil.h"
+
 static long saved_argl;
 static void *saved_argp;
 static int saved_idx;
+static int gbl_result;
 
 static void exnew(void *parent, void *ptr, CRYPTO_EX_DATA *ad,
           int idx, long argl, void *argp)
 {
-    OPENSSL_assert(idx == saved_idx);
-    OPENSSL_assert(argl == saved_argl);
-    OPENSSL_assert(argp == saved_argp);
-    OPENSSL_assert(ptr == NULL);
+    if (!TEST_int_eq(idx, saved_idx)
+      || !TEST_long_eq(argl, saved_argl)
+      || !TEST_ptr_eq(argp, saved_argp)
+      || !TEST_ptr_null(ptr))
```

Comment:
Style nit pick: || should be aligned inside the condition parenthesis, I.e. two more spaces. 
There is more than one place on this file where this needs to be done.